### PR TITLE
[channel-provider] Rewrites the wallet loader in TypeScript

### PR DESCRIPTION
**Pull request by joelalejandro**
Contributes to #277.

### CODE REVIEW

File: packages/channel-provider/src/channel-provider.ts [hunk 1]
```diff
@@ -1,15 +1,104 @@
-import EventEmitter, {ListenerFn} from 'eventemitter3';
+import debug from 'debug';
+import EventEmitter from 'eventemitter3';
+import {JsonRPCRequest} from 'web3/providers';
```
Comment (snario) [inline packages/channel-provider/src/channel-provider.ts]: Is it possible to a void using `web3`?

Comment (joelalejandro) [reply packages/channel-provider/src/channel-provider.ts]: I'm just borrowing type definitions, but sure. I could declare them on my own or use something else.

**Comment by joelalejandro** [comment]
@snario requested to add a README (with good reason!), please do not merge this yet :pray:

File: packages/channel-provider/src/channel-provider.ts (hunk 1 above)
Comment (joelalejandro) [reply packages/channel-provider/src/channel-provider.ts]: Done via 7376f32.

**Pull request merged**